google/cloud/aiplatform/datasets/dataset.py

```diff
@@ -381,93 +387,62 @@ def _create(
             request_metadata: Sequence[Tuple[str, str]] = ()
                 Strings which should be sent along with the create_dataset
                 request as metadata. Usually to specify special dataset config.
+
         Returns:
             operation (Operation):
                 An object representing a long-running operation.
         """
         gapic_dataset = gca_dataset.Dataset(
             display_name=display_name,
             metadata_schema_uri=metadata_schema_uri,
-            metadata=dataset_metadata,
+            metadata=datasource.dataset_metadata,
             labels=labels,
         )
 
         return api_client.create_dataset(
             parent=parent, dataset=gapic_dataset, metadata=request_metadata
         )
 
-    def _import_from_gcs(
-        self,
-        source: Sequence[str],
-        import_schema_uri: str,
-        data_items_labels: Optional[Dict] = None,
+    def _import_data(
+        self, datasource: datasources.NonTabularDatasourceImportable,
```

Comment:
Ditto. Depend on the abstraction, not the concrete type.

i.e. Use `_DatasourceImportable`

Basically, Datasource class should only know about the abstractions (_Datasource  and _DatasourceImportable) and not know that the concrete classes even exist. 

One exception is the Datasource.create function which we agreed to keep to maintain the current user experience.